From ffbc80ca35eb2a6d1b8d6650d70e5b44d7276c1a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:21 -0400
Subject: [ethernet] igb: Change memcpy to struct assignment

Message-id: <1409062359-13063-64-git-send-email-sassmann@redhat.com>
Patchwork-id: 89174
O-Subject: [RHEL7.1 PATCH 63/81] igb: Change memcpy to struct assignment
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Carolyn Wyborny <carolyn.wyborny@intel.com>

This patch fixes issue found by updated coccicheck.

Signed-off-by: Carolyn Wyborny <carolyn.wyborny@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a1f6347328f5ba1c64dd01ec2e9615e39be9301b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index baabc65..d8158f3 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -6499,7 +6499,7 @@ static void igb_reuse_rx_page(struct igb_ring *rx_ring,
  rx_ring->next_to_alloc = (nta < rx_ring->count) ? nta : 0;
 
  /* transfer page from old buffer to new buffer */
- memcpy(new_buff, old_buff, sizeof(struct igb_rx_buffer));
+ *new_buff = *old_buff;
 
  /* sync the buffer for use by the device */
  dma_sync_single_range_for_device(rx_ring->dev, old_buff->dma,
-- 
1.7.1